src/compiler/program.ts

```diff
@@ -1005,15 +1036,18 @@ namespace ts {
         const filesByNameIgnoreCase = host.useCaseSensitiveFileNames() ? createFileMap<SourceFile>(fileName => fileName.toLowerCase()) : undefined;
 
         if (!tryReuseStructureFromOldProgram()) {
+            forEach(rootNames, name => processRootFile(name, /*isDefaultLib*/ false));
+
             // load type declarations specified via 'types' argument
-            if (options.types && options.types.length) {
-                const resolutions = resolveTypeReferenceDirectiveNamesWorker(options.types, /*containingFile*/ undefined);
-                for (let i = 0; i < options.types.length; i++) {
-                    processTypeReferenceDirective(options.types[i], resolutions[i]);
+            const typeReferences: string[] = getDefaultTypeDirectiveNames(options, rootNames, host);
+
+            if (typeReferences) {
+                const resolutions = resolveTypeReferenceDirectiveNamesWorker(typeReferences, /*containingFile*/ undefined);
+                for (let i = 0; i < typeReferences.length; i++) {
+                    processTypeReferenceDirective(typeReferences[i], resolutions[i]);

```

Comment:
I think we want this to be useful even if you're specifying options on the commandline (or maybe I'm misunderstanding what you're getting at?)